1.0-Upgrade.md

```diff
@@ -0,0 +1,106 @@
+# kubernetes-deploy 0.30.0 -> krane 1.0 migration guide
+
+**THIS MIGRATION GUIDE IS A WORK IN PROGRESS. THINGS CAN STILL CHANGE BEFORE 1.0**
+
+The `kubernetes-deploy` became `krane` in its 1.0 version, and besides the name change we also introduced new features and [breaking changes](CHANGELOG.md). This guide will help you transition to version 1.0 as smoothly as possible.
+
+**TL;DR**:
+* The command-line interface was redesigned; besides the name change, there are breaking changes in several flags.
+* The only breaking change in the public API is the renaming of the `KubernetesDeploy` namespace to `Krane`.
+* StatsD metrics will now be generated with the `krane` prefix.
+* `kubernetes-deploy` (now `krane deploy`) / `DeployTask` can no longer deploy global (non-namespaced) resources. A new command called `krane global-deploy` and a related class called `GlobalDeployTask` were added to replace that feature.
+* If you attempt to install two gems that have conflicting executables, `gem install` will warn you but the most recently installed one will win. This means that you can run both `kubernetes-deploy` 0.30.0 and `krane` 1.0.0 side by side by doing:
+
+```bash
+gem install kubernetes-deploy
+gem install krane
+```
+
+## Public API changes
+
+The only breaking change in the public API is the renaming of the `KubernetesDeploy` namespace to `Krane`. Besides that, the API of the major public classes (`DeployTask`, `RenderTask`, `RunTask`, and `RestartTask`) didn't change at all between 0.30.0 and 1.0.0. If you're curious about this API, check the comment-based docs on these classes or the [rendered documentation at RubyGems.org](https://www.rubydoc.info/gems/kubernetes-deploy/1.0.0/KubernetesDeploy/DeployTask).
+
+## Command-line interface changes
+
+Old command | New command
+--- | --- | ---
+`kubernetes-deploy` | `krane deploy`
+`kubernetes-deploy -v` | `krane version`
+`kubernetes-render` | `krane render`
+`kubernetes-run` | `krane run`
+`kubernetes-restart` | `krane restart`
+`[kubernetes-deploy with global resources in templates]` | `krane global-deploy`
+
+### Flag changes
+
+The following tables provide a mapping of the previous flags we supported and their new version (if applicable).
```

Comment:
Should not use language `we`. Should refer to `supported in kubernetes-deploy v.X`